### PR TITLE
More random mappings

### PR DESCRIPTION
**Pull request by jaredlll08**

### CODE REVIEW

File: data/net/minecraft/core/Cursor3D.mapping [hunk 1]
```diff
@@ -0,0 +1 @@
+CLASS net/minecraft/core/Cursor3D
```
Comment (Lanse505) [inline data/net/minecraft/core/Cursor3D.mapping:1]: Why does this just add the class and no actual other stuff?
@ChampionAsh5357 @sciwhiz12 @OrionDevelopment @jaredlll08

File: data/net/minecraft/core/Vec3i.mapping [hunk 2]
```diff
@@ -32,14 +40,54 @@ CLASS net/minecraft/core/Vec3i
...
+	METHOD get (Lnet/minecraft/core/Direction$Axis;)I
+		ARG 1 axis
 	METHOD getX ()I
 		COMMENT Gets the X coordinate.
 	METHOD getY ()I
 		COMMENT Gets the Y coordinate.
 	METHOD getZ ()I
 		COMMENT Gets the Z coordinate.
+	METHOD multiply (I)Lnet/minecraft/core/Vec3i;
+		ARG 1 scalar
```
Comment (Lanse505) [inline data/net/minecraft/core/Vec3i.mapping:60]: scalar vs multiplier? both works, not sure which we want to use /cc @ChampionAsh5357

File: data/net/minecraft/data/BlockFamily.mapping [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+			ARG 1 slab
+		METHOD stairs (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 stairs
+		METHOD trapdoor (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 trapdoor
+		METHOD wall (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 wall
+	CLASS Variant
+		METHOD <init> (Ljava/lang/String;ILjava/lang/String;)V
+			ARG 3 name
```
Comment (Lanse505) [inline data/net/minecraft/data/BlockFamily.mapping:44]: shouldnt this be ARG 1 ?
@sciwhiz12

File: data/net/minecraft/sounds/SoundSource.mapping [hunk 4]
```diff
@@ -0,0 +1,3 @@
+CLASS net/minecraft/sounds/SoundSource
+	METHOD <init> (Ljava/lang/String;ILjava/lang/String;)V
+		ARG 3 name
```
Comment (Lanse505) [inline data/net/minecraft/sounds/SoundSource.mapping:3]: Once again ARG 3 should be ARG 1 ? @sciwhiz12

File: data/net/minecraft/world/entity/npc/VillagerTrades.mapping [hunk 5]
```diff
@@ -1,47 +1,145 @@
...
-	CLASS EnchantedItemForEmeralds
+		METHOD getRandomDye (Ljava/util/Random;)Lnet/minecraft/world/item/DyeItem;
+			ARG 0 random
+	CLASS EnchantBookForEmeralds
+		METHOD <init> (I)V
+			ARG 1 villagerXp
 		METHOD getOffer (Lnet/minecraft/world/entity/Entity;Ljava/util/Random;)Lnet/minecraft/world/item/trading/MerchantOffer;
 			ARG 1 trader
 			ARG 2 rand
-	CLASS ItemListing
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/npc/VillagerTrades.mapping:22]: Why did this get removed?

File: data/net/minecraft/core/Cursor3D.mapping (hunk 1 above)
Comment (marchermans) [reply data/net/minecraft/core/Cursor3D.mapping:1]: My best guess he touched the class header only but made no actual change. And to safe space enigma then only creates the file header.

File: data/net/minecraft/core/Vec3i.mapping (hunk 2 above)
Comment (marchermans) [reply data/net/minecraft/core/Vec3i.mapping:60]: I would use neither here. Factor is the more accurate term here.

File: data/net/minecraft/core/Cursor3D.mapping (hunk 1 above)
Comment (jaredlll08) [reply data/net/minecraft/core/Cursor3D.mapping:1]: I started mapping the class, realized I don't have enough information to map the method, so reverted the methods to default and I guess enigma generated the file even though nothing actually changed.

I'll remove the file in a bit

File: data/net/minecraft/world/entity/npc/VillagerTrades.mapping (hunk 5 above)
Comment (jaredlll08) [reply data/net/minecraft/world/entity/npc/VillagerTrades.mapping:22]: Seems it got rearranged to up here:
https://github.com/ParchmentMC/Parchment/pull/18/files/4c1b1d3abf611a0e837198fff444e171d8f0abbf#diff-dc2071c5ff041b39b463bc14e09b4aab0eab2e2a06477a79400a43cd367de3a3R4

File: data/net/minecraft/core/Vec3i.mapping (hunk 2 above)
Comment (Darkhax) [reply data/net/minecraft/core/Vec3i.mapping:60]: Scalar is the most correct term. This method implements scalar multiplication, where each entry in the matrix is multiplied by a scalar.

File: data/net/minecraft/data/BlockFamily.mapping (hunk 3 above)
Comment (sciwhiz12) [reply data/net/minecraft/data/BlockFamily.mapping:44]: Nope. For enum constructors, parameter `0` is the instance itself (the constructor acts as an instance method in the eyes of the JVM), parameter `1` is the implicit enum value name, parameter `2` is the ordinal of the enum value (`Enum#ordinal()`), and parameter `3` onwards are the actual declared parameters for the constructor (in this case, the `name` parameter).

File: data/net/minecraft/sounds/SoundSource.mapping (hunk 4 above)
Comment (sciwhiz12) [reply data/net/minecraft/sounds/SoundSource.mapping:3]: See https://github.com/ParchmentMC/Parchment/pull/18#discussion_r658977910.

File: data/net/minecraft/core/Vec3i.mapping (hunk 2 above)
Comment (Lanse505) [reply data/net/minecraft/core/Vec3i.mapping:60]: scalar is fine then in my opinion

File: data/net/minecraft/data/BlockFamily.mapping (hunk 3 above)
Comment (Lanse505) [reply data/net/minecraft/data/BlockFamily.mapping:44]: Okey good to know

**Review by Darkhax — DISMISSED** [review]

File: data/net/minecraft/core/GlobalPos.mapping [hunk 6]
```diff
@@ -1,4 +1,9 @@
 CLASS net/minecraft/core/GlobalPos
+	METHOD <init> (Lnet/minecraft/resources/ResourceKey;Lnet/minecraft/core/BlockPos;)V
+		ARG 1 dimension
+		ARG 2 pos
+	METHOD equals (Ljava/lang/Object;)Z
+		ARG 1 object
```
Comment (sciwhiz12) [inline data/net/minecraft/core/GlobalPos.mapping]: I've recommended `other` be used here (to contrast the `other` object against `this` object).

File: data/net/minecraft/core/Vec3i.mapping [hunk 7]
```diff
@@ -32,14 +40,54 @@ CLASS net/minecraft/core/Vec3i
...
 		COMMENT Calculate squared distance to the given Vector
-		ARG 1 to
+		ARG 1 vector
+	METHOD distSqr (Lnet/minecraft/core/Vec3i;Z)D
+		ARG 1 vector
+		ARG 2 useCenter
+	METHOD east (I)Lnet/minecraft/core/Vec3i;
+		ARG 1 distance
+	METHOD equals (Ljava/lang/Object;)Z
+		ARG 1 vector
```
Comment (sciwhiz12) [inline data/net/minecraft/core/Vec3i.mapping]: I've recommended `other` be used here (to contrast the `other` object against `this` object).

**Review by sciwhiz12 — DISMISSED** [review]
LGTM, with two slight comments (based on my previous reviews of other PRs).

**Review by jriwanek — DISMISSED** [review]

File: data/net/minecraft/core/Vec3i.mapping (hunk 7 above)
Comment (Lanse505) [reply data/net/minecraft/core/Vec3i.mapping]: I agree with Sci here however I'm gonna ping @ChampionAsh5357 for input on this.
Should "other" be standardized for equal?

File: data/net/minecraft/core/GlobalPos.mapping (hunk 6 above)
Comment (Lanse505) [reply data/net/minecraft/core/GlobalPos.mapping]: I agree with Sci here however I'm gonna ping @ChampionAsh5357 for input on this.
Should "other" be standardized for equal?

Comment (Darkhax) [reply data/net/minecraft/core/GlobalPos.mapping]: `other` is also what I have been using for comparison functions.

**Review by Lanse505 — APPROVED** [review]

**Review by jriwanek — APPROVED** [review]

**Review by Darkhax — APPROVED** [review]

**Pull request merged**